src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3182,6 +3206,82 @@ public ConstructorEdit(INamedTypeSymbol oldType)
+            if (symbol.ContainingType is not { TypeKind: TypeKind.Class or TypeKind.Struct })
+                return false;
+
+            // We store the containing symbol in NewSymbol of the edit for later use.
+            if (symbol is IMethodSymbol
+                {
+                    MethodKind:
+                        MethodKind.Ordinary or
+                        MethodKind.Constructor or
+                        MethodKind.EventAdd or
+                        MethodKind.EventRemove or
+                        MethodKind.EventRaise or
+                        MethodKind.Conversion or
+                        MethodKind.UserDefinedOperator or
+                        MethodKind.PropertyGet or
+                        MethodKind.PropertySet
+                })
+            {
+                return true;
+            }
+
+            return symbol is IPropertySymbol or IEventSymbol;
+        }
+
+        /// <summary>
+        /// Add semantic edits for the specified symbol, or the associated members of the specified symbol,
+        /// for example, edits for each accessor if a property symbol is passed in.
+        /// </summary>
+        private static void AddMemberOrAssociatedMemberSemanticEdits(ArrayBuilder<SemanticEditInfo> semanticEdits, SemanticEditKind editKind, ISymbol symbol, SymbolKey? containingSymbolKey, Func<SyntaxNode, SyntaxNode?>? syntaxMap, SymbolKey? partialType, CancellationToken cancellationToken)
+        {
+            Debug.Assert(symbol is IMethodSymbol or IPropertySymbol or IEventSymbol);
+
+            // We store the containing symbol in NewSymbol of the edit for later use.
+            if (symbol is IMethodSymbol)
+            {
+                AddEdit(semanticEdits, editKind, symbol, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+            }
+            else if (symbol is IPropertySymbol propertySymbol)
+            {
+                AddEdit(semanticEdits, editKind, propertySymbol.GetMethod, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+                AddEdit(semanticEdits, editKind, propertySymbol.SetMethod, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+            }
+            else if (symbol is IEventSymbol eventSymbol)
+            {
+                AddEdit(semanticEdits, editKind, eventSymbol.AddMethod, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+                AddEdit(semanticEdits, editKind, eventSymbol.RemoveMethod, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+                AddEdit(semanticEdits, editKind, eventSymbol.RaiseMethod, containingSymbolKey, syntaxMap, partialType, cancellationToken);
+            }
+
+            static void AddEdit(ArrayBuilder<SemanticEditInfo> semanticEdits, SemanticEditKind editKind, ISymbol? symbol, SymbolKey? containingSymbolKey, Func<SyntaxNode, SyntaxNode?>? syntaxMap, SymbolKey? partialType, CancellationToken cancellationToken)
```

Comment:
It would be simpler to make this local function non-static and only take `symbol` parameter.